### PR TITLE
Barcode

### PR DESCRIPTION
**Pull request by rishabhraaj17**
## Issue

This fixes the following issue(s):

### CODE REVIEW

**Comment by rishabhraaj17** [comment]
After scanning the QRCode, user joins the chat room, but if we press the back key it shows the reloading symbol and chat room list is not updated instantly,  it keeps reloading. If we again press back key twice the chat list then shows this newly added course. I understand this is a problem that needs a fix, but I wanted to show that now after scanning user joins the chat room. If I am wrong anywhere or misunderstood anything please let me know.

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/chat/activity/ChatRoomsActivity.java [hunk 1]
```diff
@@ -155,14 +155,20 @@ private void startJoinRoom() {
 
     @Override
     protected void onActivityResult(int requestCode, int resultCode, Intent data) {
-        if (requestCode == 1 && resultCode == RESULT_OK && data != null) {
+        Utils.log("onActivity is this ");
+        if (resultCode == RESULT_OK && data != null) {   //removed requestCode; but maybe inspite of requestCode = 1, we need something else.
```
Comment (jacqueline8711) [inline app/src/main/java/de/tum/in/tumcampusapp/component/ui/chat/activity/ChatRoomsActivity.java]: it should be requestCode == JOIN_ROOM_REQUEST_CODE (the constant that is given to the startActivityForResult in the method above)

File: .idea/codeStyleSettings.xml [hunk 2]
```diff
@@ -1,264 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (kordianbruck) [inline .idea/codeStyleSettings.xml]: Please don't delete these files in .idea

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/chat/activity/ChatRoomsActivity.java [hunk 3]
```diff
@@ -155,14 +155,20 @@ private void startJoinRoom() {
...
             if (name.charAt(3) == ':') {
+                Utils.log("about to call createorjoinchat room");
                 createOrJoinChatRoom(name);
             } else {
                 Utils.showToast(this, R.string.invalid_chat_room);
             }
         }
+        //String name = data.getStringExtra("name");
+        //Utils.log("name is this " +name);
+        //createOrJoinChatRoom(name);
```
Comment (kordianbruck) [inline app/src/main/java/de/tum/in/tumcampusapp/component/ui/chat/activity/ChatRoomsActivity.java]: Please never commit commented code. Thats what we have a VCS for.

**Comment by kordianbruck** [comment]
Closing due to inactivity.

**Pull request closed without merging**